File: Aplicación Web con React/components/BookForm.js
import React, { useState, useEffect } from 'react';

function BookForm({ addBook, updateBook, existingBook }) {
    const [book, setBook] = useState({
        title: '',
        author: '',
        price: '',
    });

    useEffect(() => {
        if (existingBook) {
            setBook(existingBook);
        }
    }, [existingBook]);

    const handleChange = (e) => {
        const { name, value } = e.target;
        setBook({ ...book, [name]: value });
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if (book.title && book.author && book.price) {
            if (existingBook) {
                updateBook(book);
            } else {
                addBook(book);
            }
            setBook({ title: '', author: '', price: '' });
        }
    };

    return (
        <div className="book-form">
            <h2>{existingBook ? 'Editar Libro' : 'Agregar Libro'}</h2>
            <form onSubmit={handleSubmit}>
                <label>
                    Título:
                    <input
                        type="text"
                        name="title"
                        value={book.title}
                        onChange={handleChange}
                        required
                    />
                </label>
                <label>
                    Autor:
                    <input
                        type="text"
                        name="author"
                        value={book.author}
                        onChange={handleChange}
                        required
                    />
                </label>
                <label>
                    Precio:
                    <input
                        type="number"
                        name="price"
                        value={book.price}
                        onChange={handleChange}
                        required
                    />
                </label>
                <button type="submit">{existingBook ? 'Actualizar' : 'Agregar'}</button>
            </form>
        </div>
    );
}

export default BookForm;

File: REFLEX/bookform.py
from reflex import Reflex, html, use_state

def BookForm(add_book=None, update_book=None, book=None):
    title = use_state(book["title"] if book else "")
    author = use_state(book["author"] if book else "")
    price = use_state(book["price"] if book else "")

    def submit():
        if book:
            update_book({"id": book["id"], "title": title, "author": author, "price": price})
        else:
            add_book({"title": title, "author": author, "price": price})

    return html.form({"class": "book-form"}, [
        html.label("Título"),
        html.input(type="text", value=title, on_change=title.set),

        html.label("Autor"),
        html.input(type="text", value=author, on_change=author.set),

        html.label("Precio"),
        html.input(type="number", value=price, on_change=price.set),

        html.button("Guardar", on_click=submit)
    ])
